app.py

```diff
@@ -0,0 +1,60 @@
+import base64
+import binascii
+import hashlib
+import hmac
+import urllib.parse
+
+from flask import abort, Flask, redirect, request
+
+app = Flask(__name__)
+
+try:
+    SECRET = open('keys/secret').read().strip()
+except FileNotFoundError:
+    print('No secret file found')
```

Comment:
Do we also plan to die if no secret file is found?

Reply:
I'd like to keep the `/test` endpoint active even if the secret file doesn't exist. But perhaps we could control this with an environment variable, to cause this to fail early in production.

Reply:
I opine that `SECRET` should be defined at all times. Otherwise you'll have a recipe for `NameError`s.

Reply:
You're probably right. I'll change this to terminate if there's no SECRET, and modify the Makefile to make an empty secret file in development mode.the